lib/Basics/files.cpp

```diff
@@ -455,7 +455,7 @@ int TRI_ChMod(char const* path, long mode, std::string& err) {
   if (res != 0) {
     err = "error setting desired mode " + std::to_string(mode) + " for file " +
           path + ": " + strerror(errno);
-    return errno;
+    return TRI_set_errno(TRI_ERROR_SYS_ERROR);
```

Comment:
This was and is potentially unsafe to do.
We are referring to errno here _after_ calling std::to_string() and strerror().
But at least strerror() may clobber the value in errno:

From `man 3 strerror`:
```
        POSIX.1-2001 and POSIX.1-2008 require that a successful  call  to  str‐
       error()  or  strerror_l()  shall  leave errno unchanged, and note that,
       since no function return value is reserved to indicate an error, an ap‐
       plication  that  wishes  to check for errors should initialize errno to
       zero before the call, and then check errno after the call.
```

So it would be better to first capture errno and store the result of `TRI_set_errno(TRI_ERROR_SYS_ERROR)` somewhere, and only then build the string and call strerror().

Reply:
Good catch, thanks. Fixed.